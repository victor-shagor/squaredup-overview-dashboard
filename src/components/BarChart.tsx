import Chart from "react-apexcharts";
import { ChartProps } from "constants/interfaces";

const BarChart = ({ series, options, width, height }: ChartProps) => {
  return (
    <div data-testid="barchart-container">
      <Chart
        options={{ chart: { type: "bar" }, ...options }}
        type="bar"
        width={width}
        height={height}
        series={series}
      />
    </div>
  );
};
export default BarChart;
