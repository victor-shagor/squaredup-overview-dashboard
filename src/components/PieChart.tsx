import React from "react";

import Chart from "react-apexcharts";
import { ChartProps } from "constants/interfaces";

const PieChart = ({ options, series, width, height }: ChartProps) => {
  return (
    <div data-testid="piechart-container">
      <Chart
        data-testid="piechart"
        options={{
          legend: {
            position: "right",
            offsetY: 100,
            height: 300,
          },
          ...options,
        }}
        type="donut"
        width={width}
        height={height}
        series={series}
      />
    </div>
  );
};

export default PieChart;
