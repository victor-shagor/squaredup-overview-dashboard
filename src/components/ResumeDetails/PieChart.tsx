import React from "react";

import Chart from "react-apexcharts";
import { PieChartProps } from "constants/interfaces";

const PieChart = ({ label, colors, series }: PieChartProps) => {
  const options = {
    labels: label,
    colors: colors,
    dataLabels: {
      enabled: false,
    },

    plotOptions: {
      pie: {
        donut: {
          size: "58%",
          labels: {
            show: true,
            name: {
              show: true,
              fontSize: "22px",
              fontFamily: "Helvetica, Arial, sans-serif",
              fontWeight: 600,
              color: undefined,
              offsetY: 10,
            },
            value: {
              show: true,
              fontSize: "16px",
              fontWeight: "bold",
              color: undefined,
              offsetY: -20,
            },
            total: {
              show: true,
              showAlways: true,
              label: "Total Resumes",
              fontSize: "15px",
              fontFamily: "SFProText",
              color: "#373d3f",
            },
          },
        },
      },
    },
  };
  return (
    <div>
      <Chart
        options={{
          legend: {
            position: "right",
            offsetY: 100,
            height: 300,
          },
          ...options,
        }}
        type="donut"
        width={500}
        height={320}
        series={series}
      />
    </div>
  );
};

export default PieChart;
