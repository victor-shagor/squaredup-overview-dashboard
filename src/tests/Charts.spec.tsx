import { render, screen } from "@testing-library/react";
import BarChart from "components/BarChart";
import PieChart from "components/PieChart";
import renderer from "react-test-renderer";

jest.mock("react-apexcharts", () => {
  return {
    __esModule: true,
    default: () => {
      return <div />;
    },
  };
});

const options = {
  labels: ["test1", "test2"],
  colors: ["#5a5bb5", "#5dace6"],
  dataLabels: {
    enabled: false,
  },
};
const series = [30, 70];

const mockData = [
  {
    name: "Guest Users",
    data: [225, 1419, 1665, 194, 3, 3, 3, 2],
  },
  {
    name: "Verified Users",
    data: [7, 329, 451, 643, 310, 559, 567, 115],
  },
];

describe("Charts", () => {
  it("should render piechart component", () => {
    render(
      <PieChart options={options} width={500} height={320} series={series} />
    );
    expect(screen.getByTestId("piechart-container")).toBeInTheDocument();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <PieChart options={options} width={500} height={320} series={series} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render BarChart component", () => {
    render(
      <BarChart options={options} width={500} height={320} series={mockData} />
    );
    expect(screen.getByTestId("barchart-container")).toBeInTheDocument();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <BarChart
          options={options}
          width={500}
          height={320}
          series={mockData}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
