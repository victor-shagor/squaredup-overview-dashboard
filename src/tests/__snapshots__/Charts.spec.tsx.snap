// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charts renders correctly 1`] = `
<div
  data-testid="piechart-container"
>
  <div />
</div>
`;

exports[`Charts renders correctly 2`] = `
<div
  data-testid="barchart-container"
>
  <div />
</div>
`;
